### PR TITLE
basic go support

### PR DESCRIPTION
**Pull request by nanu-c**

### CODE REVIEW

File: clickable [hunk 1]
```diff
@@ -1093,7 +1099,35 @@ class PythonClickable(PureClickable):
...
     pass
 
+class GoClickable(Clickable):
+    def go_install(self):
+        get_output=True
+        if not os.path.exists(self.temp):
+            os.makedirs(self.temp)
+        gocommand = "/usr/local/go/bin/go build -i -o {}/{} .".format(self.temp,
+                                                                    self.config.app)
+        wrapped_command = 'docker run -v {}:{} -v {}:/home/developer/gopath -w {} -u {} --rm -i {} bash -c "{}"'.format(
```
Comment (bhdouglass) [inline clickable]: You should use run_container_command here, so you don't need to do all the docker setup

File: clickable [hunk 2]
```diff
@@ -1093,7 +1099,35 @@ class PythonClickable(PureClickable):
     # The only difference between this and the Pure template is that this doesn't force the "all" arch
     pass
 
+class GoClickable(Clickable):
+    def go_install(self):
+        get_output=True
+        if not os.path.exists(self.temp):
+            os.makedirs(self.temp)
```
Comment (bhdouglass) [inline clickable]: This should already exist, not need to do it again here.

File: clickable [hunk 3]
```diff
@@ -1093,7 +1099,35 @@ class PythonClickable(PureClickable):
     # The only difference between this and the Pure template is that this doesn't force the "all" arch
     pass
 
+class GoClickable(Clickable):
+    def go_install(self):
```
Comment (bhdouglass) [inline clickable]: Any reason this isn't just in the _build() function?

File: docker/ubuntu-sdk/15.04-armhf/Dockerfile [hunk 4]
```diff
@@ -34,8 +34,18 @@ RUN apt-get -y --force-yes --no-install-recommends install \
     qt5-doc \
     language-pack-en \
     click
+# go dependencies
+RUN apt-get -y --force-yes --no-install-recommends install \
```
Comment (bhdouglass) [inline docker/ubuntu-sdk/15.04-armhf/Dockerfile]: Go ahead and combine this with the previous line, not need to have the packages installed in separate lines.

**Comment by bhdouglass** [comment]
This looks like a great start, I can't wait to see more!

File: clickable (hunk 1 above)
Comment (nanu-c) [reply clickable]: For go I need to mount the gopath and the run_container_command is only able to mount one volume

Comment (nanu-c) [reply clickable]: Also I need to fix that the go compile doesn't need root.

Comment (bhdouglass) [reply clickable]: @nanu-c you could update the run_container_command function to add that volume if a go template is being used.

**Comment by nanu-c** [comment]
I don't know if it's worth to finish this, because there are 
2 implementations of qml/qtquick and go
* https://github.com/therecipe/qt/ -> is maintained, doesn't support ubuntu touch and includes also the build system, if that would work, we don't need a clickable implementation
* https://github.com/go-qml/qml isn't supported since 2016 and doesn't support go >1.5, the best fork of them https://github.com/amlwwalker/qml works with go 1.6 but is broken with QtQuick 2.4 and Ubuntu.Components 1.3, the most go-qml forks disabled the issue tool here on github, and they are also not explicitly made for ubuntu touch. Actually the findChildren isn't working and think it's somehow because the lib use QQuickEngine and not QQuickApplicationEngine. But I started my first qml project in december so it's way to much time to learn (for me qml is not so intuitive) and I don't want to dig in the stomach of qtquick.

The signal app is written in go and for maintaining it I have to maintain
* the signal-crypto-mechanism for go
* the api communication for signal with the signal server in go
* the bridge between qml and go

Is there another developer with experience in go + qml on ubuntu touch?

For this pull request to be complete the following manual steps have to be included

i use this script

```
#!/bin/bash

mkdir -p ../build/tmp/qml
cp -a ../qml/phoneui ../build/tmp/qml
cp -a ../docs/CHANGELOG.md ../build/tmp
cp -a ../click/* ../build/tmp
cp -a ../lib ../build/tmp

# Build and include translations
for po in ../po/*.po; do
	loc=$(echo $(basename $po)|cut -d'.' -f1)
	dir=../build/tmp/share/locale/$loc/LC_MESSAGES
	mkdir -p $dir
	msgfmt $po -o $dir/textsecure.nanuc.mo
done
```

and 

```
CLICK_NAME=textsecure.nanuc*click

adb push $CLICK_NAME /home/phablet
adb shell pkcon install-local $CLICK_NAME --allow-untrusted
clickable -s UA000784 launch logs
```

**Comment by bhdouglass** [comment]
Unfortunately I don't have any experience with go & qml, just with go for my Falcon scope. I like the idea of being able to easily use go & qml for Ubuntu Touch, it really expands developer options. 

With the first option, couldn't you just include that in your go path and compile it from within clickable kinda like you are doing here?

File: clickable (hunk 1 above)
Comment (nanu-c) [reply clickable]: I did it, only problem, still need user 0

File: clickable (hunk 2 above)
Comment (nanu-c) [reply clickable]: done

File: clickable (hunk 3 above)
Comment (nanu-c) [reply clickable]: Changed this

File: docker/ubuntu-sdk/15.04-armhf/Dockerfile (hunk 4 above)
Comment (nanu-c) [reply docker/ubuntu-sdk/15.04-armhf/Dockerfile]: Changed it

**Comment by nanu-c** [comment]
My only problem when I run `clickable` without parameters is this:

`
[Errno 2] No such file or directory: '/home/nanu/src/go/src/github.com/nanu-c/textsecure-qml/build/textsecure.nanuc_0.3.21_armhf.click'`

It's actually in 
`'/home/nanu/src/go/src/github.com/nanu-c/textsecure-qml/textsecure.nanuc_0.3.21_armhf.click'`

**Comment by bhdouglass** [comment]
@nanu-c I just merged tis in with a few of my own fixes: https://github.com/bhdouglass/clickable/commit/ef497ac9e430d3cc6ca472f5ffc138e9a941ceca
I was able to get it to not require a root user and fixed the click being in the wrong directory.

**Pull request merged**